tests/models/test_dag.py

```diff
@@ -2234,3 +2234,40 @@ def next_dagrun_info(self, last_automated_data_interval, restriction):
         ),
     ]
     assert caplog.records[0].exc_info is not None, "should contain exception context"
+
+
+@pytest.mark.parametrize(
+    "logical_date, data_interval_start, data_interval_end, expected_data_interval",
+    [
+        pytest.param(None, None, None, None, id="no-next-run"),
+        pytest.param(
+            DEFAULT_DATE,
+            DEFAULT_DATE,
+            DEFAULT_DATE + timedelta(days=1),
```

Comment:
for modern, your explicit data interval and the schedule interval are both 1 day.  so it's sortof ambiguous what's driving the result.  seems it might be better to choose an interval that differs from the schedule interval of the dag, to make it really clear what drives the result

Reply:
Good point. It is logically not possible for a DAG to return different data intervals via inference and explicit timetable (for obvious reasons), but it is a good way to test the behaviour in a unit test.